templates/web/layout.html

```diff
@@ -48,18 +48,21 @@
 
             <div class="collapse navbar-collapse" id="navbarSupportedContent">
               <ul class="navbar-nav me-auto">
+                {% url 'student_index' as student_index_url %}
+
                 {% if is_teacher %}
-                <li class="nav-item"><a class="nav-link" href="{% url 'submits' %}">Submits</a></li>
-                <li class="nav-item"><a class="nav-link" href="{% url 'tasks' %}">Tasks</a></li>
-                {% endif %}
-                {% if is_teacher %}
-                <li class="nav-item"><a class="nav-link" href="{% url 'student_index' %}">Student view</a></li>
-                {% if user.is_staff %}
-                {% comment %}
-                https://stackoverflow.com/questions/694477/getting-django-admin-url-for-an-objects
-                {% endcomment %}
-                <li class="nav-item"><a class="nav-link" href="{% url 'admin:index' %}">Admin</a></li>
-                {% endif %}
+                  <li class="nav-item"><a class="nav-link" href="{% url 'submits' %}">Submits</a></li>
+                  <li class="nav-item"><a class="nav-link" href="{% url 'tasks' %}">Tasks</a></li>
+                  {% if request.path != student_index_url %}
+                    <li class="nav-item"><a class="nav-link" href="{% url 'student_index' %}">Student view</a></li>
+                  {% endif %}
+                {%  endif %}
+
+                {% if is_teacher and user.is_staff %}
```

Comment:
`is_teacher` state is being checked once before this. I think this could be nested in the previous `if` to make it a bit simpler.

I tested your change locally and it works as expected :+1:

Reply:
OK, thanks for spotting this.